### ISSUE TITLE
[webview_flutter] Android Keyboard automatically closed in landscape mode

### ISSUE THREAD

**Issue by hnvn** [flutter/flutter#56513]
In Android landscape mode, input field cannot be edited because the keyboard is automatically closed as soon as it shows up.

Here's the result when I run the example project in my Xperia device (Android 7.1.1)

![Farmers Market Finder Demo](https://media.giphy.com/media/ZDFOeacvUWiAbXARWU/giphy.gif)

flutter doctor:
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel beta, v1.17.0, on Mac OS X 10.15.4 19E287, locale en-US)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.0-rc2)
[✓] Xcode - develop for iOS and macOS (Xcode 11.4)
[✓] Chrome - develop for the web
[✓] Android Studio (version 3.6)
[✓] VS Code (version 1.44.2)
[✓] Connected device (3 available)
```

**Comment by hnvn** [comment]
I can temporarily workaround it by set `resizeToAvoidBottomInset=false` but it leads to other problem that the keyboard overlapping the input fields

**Comment by slightfoot** [comment]
The input field is larger than the available space. Can you try making a demo html page that has an input field that is smaller and see if it replicates

**Comment by hnvn** [comment]
I don't think it's the matter of input field size. As a further investigation, I can keep the keyboard on the screen (with `resizeToAvoidBottomInset=true`) if I place a bottom margin to the webview. In my device, a bottom margin value at least 80.0 (the height of the keyboard, in this case, is 200.0) will keep the keyboard on the screen, otherwise, the keyboard will be automatically closed.

**Comment by darshankawar** [comment]
I am able to replicate this issue on `dev channel (1.20.0-0.0.pre)`. Used below code sample to replicate and tested on Emulator Pixel 3XL, api 27.
In landscape mode, when tapped on input field, the keyboard opens and closes again.

<details>
<summary>Code Sample</summary>

```
import 'dart:convert';
import 'package:flutter/material.dart';
import 'package:webview_flutter/webview_flutter.dart';

void main() {
  runApp(MyApp());
}

const String kNavigationExamplePage = '''
<!DOCTYPE html><html>
<head><title>Navigation Delegate Example</title></head>
<body>
<p>
The navigation delegate is set to block navigation to the youtube website.
</p>

<input type='search'  placeholder="search"/>
<input type='password'  placeholder="password" />
<input type ='month'  />
<input type = 'file' />
</body>
</html>
''';

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
        visualDensity: VisualDensity.adaptivePlatformDensity,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);
  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  WebViewController _webViewController;
  String filePath = "data/user_protocol.html";

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text("html"),
      ),
      body: Container(
        child: WebView(
          initialUrl: "",
          onWebViewCreated: ((WebViewController vc) {
            _webViewController = vc;
            _loadlocalData();
          }),
        ),
      ),
    );
  }

  _loadlocalData() async {
    final String contentBase64 =
        base64Encode(const Utf8Encoder().convert(kNavigationExamplePage));
    _webViewController.loadUrl('data:text/html;base64,$contentBase64');
  }
}

```
</details>


<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel dev, 1.20.0-0.0.pre, on Mac OS X 10.15.2 19C57, locale en-US)
    • Flutter version 1.20.0-0.0.pre at /Users/deeptibelsare/Documents/Fluttersdk/flutter
    • Framework revision d9653445f4 (8 days ago), 2020-06-09 18:43:03 -0400
    • Engine revision e8c13aa012
    • Dart version 2.9.0 (build 2.9.0-14.0.dev 5c1376615e)

 
[!] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/deeptibelsare/Library/Android/sdk
    • Platform android-29, build-tools 29.0.3
    • ANDROID_HOME = /Users/deeptibelsare/Library/Android/sdk
    • ANDROID_SDK_ROOT = /Users/deeptibelsare/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
    ✗ Android license status unknown.
      Try re-installing or updating your Android SDK Manager.
      See https://developer.android.com/studio/#downloads or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions.

 
[✓] Xcode - develop for iOS and macOS (Xcode 11.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3, Build version 11C29
    • CocoaPods version 1.9.3

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 45.1.1
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)

[✓] Connected device (1 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 8.1.0 (API 27) (emulator)

! Doctor found issues in 1 category.

```
</details>

**Comment by blasten** [comment]
This has been fixed in Flutter 1.22/ WebView 1.0 - use hybrid composition in the Flutter WebView plugin.

https://github.com/flutter/plugins/tree/master/packages/webview_flutter#using-hybrid-composition
